core/deco.c

```diff
@@ -477,11 +477,12 @@ void calc_crushing_pressure(struct deco_state *ds, double pressure)
 }
 
 /* add period_in_seconds at the given pressure and gas to the deco calculation */
-void add_segment(struct deco_state *ds, double pressure, const struct gasmix *gasmix, int period_in_seconds, int ccpo2, const struct dive *dive, int sac)
+bool add_segment(struct deco_state *ds, double pressure, const struct gasmix *gasmix, int period_in_seconds, int ccpo2, const struct dive *dive, int sac)
```

Comment:
Encoding the icd warning as return value isn't the prettiest solution. What about adding a icd_warning to the struct deco_state instead?

Or even turning it into its own helper, which just checks for the icd warning conditions. Maybee its just a huge overlap with what's already happening in add_segment...

Reply:
including it in deco_state is actually a good idea. Will do that.

I had started out writing an independent helper function but then realised that so much of the calculation is the same as done in add_segment. So in order in minimise the computational footprint I went for the solution of placing it there.